components/admin/AdminShell.tsx

```diff
@@ -55,12 +60,20 @@ export default function AdminShell({
 }: PropsWithChildren<Props>) {
   const [opened, { toggle }] = useDisclosure();
   const pathname = usePathname();
+  const [comp, setComp] = useState<string | null>(null);
 
   const { data } = useSWR<Competition[]>('/api/comp', fetcher, {
     fallbackData: [],
   });
 
-  const [comp, setComp] = useState<string | null>(null);
+  useEffect(() => {
+    const selectedComp = sessionStorage.getItem('selectedComp');
+
+    // Check if the selected competition still exists
+    if (selectedComp && data?.find((c) => c.code === selectedComp)) {
```

Comment:
We don't care about the value here. We only need to know if there exists some item that matches the predicate. Use `.some(...)`.

Reply:
sounds good :+1: